dart/lib/src/client.dart

```diff
@@ -183,29 +184,35 @@ abstract class SentryClient {
     );
 
     if (response.statusCode != 200) {
-      var errorMessage = 'Sentry.io responded with HTTP ${response.statusCode}';
+      /*var errorMessage = 'Sentry.io responded with HTTP ${response.statusCode}';
       if (response.headers['x-sentry-error'] != null) {
         errorMessage += ': ${response.headers['x-sentry-error']}';
-      }
-      return SentryResponse.failure(errorMessage);
+      }*/
+      return SentryId.empty();
     }
 
-    final eventId = '${json.decode(response.body)['id']}';
-    return SentryResponse.success(eventId: eventId);
+    final eventId = json.decode(response.body)['id'];
+    return eventId != null ? SentryId(eventId) : SentryId.empty();
   }
 
-  /// Reports the [exception] and optionally its [stackTrace] to Sentry.io.
-  Future<SentryResponse> captureException({
-    @required dynamic exception,
-    dynamic stackTrace,
-  }) {
+  /// Reports the [throwable] and optionally its [stackTrace] to Sentry.io.
+  Future<SentryId> captureException(dynamic throwable, {dynamic stackTrace}) {
     final event = Event(
-      exception: exception,
+      exception: throwable,
       stackTrace: stackTrace,
     );
     return captureEvent(event: event);
   }
 
+  /// Reports the [message]
+  Future<SentryId> captureMessage({
+    @required Message message,
+    SeverityLevel level = SeverityLevel.warning,
```

Comment:
default is `info`

Reply:
is there an implicit convert between `Message` and `String`?
It should be possible to capture:

`Sentry.captureMessage("string here")` but I'm afraid that with our `Message` class it'll need to be:

`Sentry.captureMessage({message: Message({formatted: "string here"})})` which isn't really straight forward.

Reply:
we could declare `Sentry.captureMessage(String message,{ List<dynamic> messageParams = [], SeverityLevel level })` and instanciate a `Message' in the hub with only a `formatted` value if `params.isEmpty` ?

Reply:
as overloading is not supported by Dart, I suggest using all of them as optional named params.
captureMessage({formatted, message, params}) so they choose what to pass, either formatted OR message + params, the docs on top of the method should be clear about it

Reply:
The snippet from Erick would work thouhg, the first arg would be positional, and the following would be named. That would work, right?
I looks good to me since you can:

`Sentry.captureMessage("bla")`;
and

`Sentry.captureMessage("bla", {level: SentryLevel.debug});`

Not sure about params though. The `formatted` stuff is more useful for logging libraries.
I'm unaware of folks using it with Sentry.captureMessage. but maybe it makes sense to take into account here. but in this case the positional argument would be `formatted` (although doesn't have to be named that way), and the named arg could take the template:

```dart
Sentry.captureMessage("message with user id 1", {
level: SentryLevel.debug,
params: ["1"],
template: "message with user id %s"
});
```

Looks pretty horrible though.

Reply:
I'm not sure I understand how Message will be used. Will it be instantiated with formatted AND template AND params ?

Reply:
Minimum you'd send `formatted`, with the interpolated string, which would show in Sentry the right value. Optionally you'd send `message` which would be used in grouping so that:

"user %s logged in" would all group together, instead of "user 1 logged in" and "user 2 logged in" becomes two separate issues in Sentry. But the latter is useful since when debugging you want to know the actual value. That's the difference of formatted and message.

Params is optional, but it can show in the UI the individual arguments, so it's nice to include.

This original was called "logEntry" since it's designed to be compatible with a logging library, such as you write:

logger.warn("user %s", 1); and under the hood we can replace the %s with 1 and get the 3 things (formatted, message and params).

But for captureMessage, it should be a simple API to send a message. Optional level is nice to have too because by default it's `info`. Having captureMessage take a class, makes it harder to send a basic string (unless we have some implicit operator in Dart). 

I propose we keep what you did here except:

```dart
 Future<SentryId> captureMessage(
    @required String message,
    {SeverityLevel level = SeverityLevel.info}) {
final event = Event({
message: Message(formatted: message, level: level)}
captureEvent(event);
}
```

I'd also prose having all these method become extension methods and not part of the actual interface.

Someone implementing a client shoulndn't have to make these choices. The interface could be much simpler, for example: [ISentryClient](https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry/ISentryClient.cs) and [extensions](https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry/SentryClientExtensions.cs#L33-L47)"

Reply:
I could suggest this approach in Java because Java 1.8 (or any?) has extension method, but Dart does.

Reply:
extension methods in Dart are only available in Dart 2.7, is this realistic enough to add @rxlabz or people still use older versions than that?
got @bruno-garcia point, I thought that when you have `message` and `params`, Sentry would set the `formatted` field automatically, but yeah, `message` and `params` should be named params then

Reply:
what does it happen if someone is using Dart < 2.7? does the SDK wont work?

Reply:
yes, to use the library you'll have to install the min sdk

Reply:
Update : now `hub.captureMessage` and `client.captureMessage` only take a String message required argument, and an optional named `level`.
```dart
Future<SentryId> captureMessage(String message , { SeverityLevel level = SeverityLevel.info});
```

Reply:
@rxlabz whats about the `params` and `template`? they should also be named params

Reply:
> I propose we keep what you did here except:

 > Future<SentryId> captureMessage(
    @required String message,
    {SeverityLevel level = SeverityLevel.info}) {
final event = Event({
message: Message(formatted: message, level: level)}
captureEvent(event);
}

I thought that was the final decision. So now we have :
```dart
Future<SentryId> captureMessage(
    String formatted, {
    SeverityLevel level = SeverityLevel.info,
    String template,
    List<dynamic> params,
  }) {
    final event = Event(
      message: Message(formatted, template: template, params: params),
      level: level,
    );
    return captureEvent(event);
  }
```

Reply:
nice, only `formatted` is mandatory anyway so I'm in for that